controllers/k8ssandra/k8ssandracluster_controller_test.go

```diff
@@ -1195,6 +1198,114 @@ func createMultiDcClusterWithStargate(t *testing.T, ctx context.Context, f *fram
 	verifyObjectDoesNotExist(ctx, t, f, sg2Key, &stargateapi.Stargate{})
 }
 
+// changeNumTokensValue creates a Datacenter and then changes the numTokens value
+// Such change is prohibited and should fail
+func changeNumTokensValue(t *testing.T, ctx context.Context, f *framework.Framework, namespace string) {
+	require := require.New(t)
+
+	k8sCtx := "cluster-1"
+
+	kc := &api.K8ssandraCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      "test",
+		},
+		Spec: api.K8ssandraClusterSpec{
+			Cassandra: &api.CassandraClusterTemplate{
+				Cluster: "test",
+				CassandraConfig: &api.CassandraConfig{
+					CassandraYaml: api.CassandraYaml{
+						NumTokens: pointer.Int(16),
+					},
+				},
+				Datacenters: []api.CassandraDatacenterTemplate{
+					{
+						Meta: api.EmbeddedObjectMeta{
+							Name: "dc1",
+						},
+						K8sContext:    k8sCtx,
+						Size:          1,
+						ServerVersion: "3.11.10",
+						StorageConfig: &cassdcapi.StorageConfig{
+							CassandraDataVolumeClaimSpec: &corev1.PersistentVolumeClaimSpec{
+								StorageClassName: &defaultStorageClass,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	err := f.Client.Create(ctx, kc)
```

Comment:
This creates the K8ssandraCluster in `cluster-0` but also note that the CassandraDatacenter will be created in `cluster-1`.

Reply:
`f.Client` always refers to `cluster-0`, i.e., the control plane cluster.

Reply:
ooooh, I see. Things make sense now 😅